### PR TITLE
chore(gatsby-plugin-offline): add pluginOptionsSchema export 

### PR DESCRIPTION
**Pull request by mfrachet**
## Description

Add plugin schema options in `gatsby-plugin-offline`

Relates to #27242 and https://www.gatsbyjs.com/plugins/gatsby-plugin-offline/

### CODE REVIEW

File: packages/gatsby-plugin-offline/src/gatsby-node.js [hunk 1]
```diff
@@ -211,3 +211,47 @@ exports.onPostBuild = (
...
+        runtimeCaching: Joi.array().items(
+          Joi.object({
+            urlPattern: Joi.object().instance(RegExp),
+            handler: Joi.string().valid(
+              `StaleWhileRevalidate`,
+              `CacheFirst`,
+              `NetworkFirst`,
+              `NetworkOnly`,
+              `CacheOnly`
+            ),
```
Comment (mfrachet) [inline packages/gatsby-plugin-offline/src/gatsby-node.js:247]: TIL: this is the way to deal with "enums" like options

File: packages/gatsby-plugin-offline/src/__tests__/gatsby-node.js [hunk 2]
```diff
@@ -107,3 +109,82 @@ describe(`onPostBuild`, () => {
     expect(swText).toContain(`debug: true`)
   })
 })
+
+describe.only(`pluginOptionsSchema`, () => {
```
Comment (mfrachet) [inline packages/gatsby-plugin-offline/src/__tests__/gatsby-node.js:113]:
```suggestion
describe(`pluginOptionsSchema`, () => {
```

File: packages/gatsby-plugin-offline/src/gatsby-node.js (hunk 1 above)
Comment (mxstbr) [reply packages/gatsby-plugin-offline/src/gatsby-node.js:247]: We should definitely add this to the best practices documentation we're going to write!

**Review by mxstbr — APPROVED** [review]
Amazing, thanks so much for tackling this!

**Pull request merged**